### DISCUSSION TITLE
afl-proxy. Can't reach while loop in main.

### DISCUSSION THREAD

**Discussion by sk3pta** [AFLplusplus/AFLplusplus#2028] [Q&A] [answered]
Hi,
I am trying the way I can use afl-proxy in order to expand AFL++ for fuzz embedded device without any binary nor debug interface and the problem is that the while loop condition in main is always false.
```c
__afl_start_forkserver();
  llogf("Before while\n"); // logs to file
  while ((len = __afl_next_testcase(buf, sizeof(buf))) > 0) {
    llogf("inside while\n"); // logs to file
```

`sudo afl-fuzz -i ./input/ -o out/ -- ./afl-proxy`
as a result, after afl-fuzz , there is only "Before while" in my log file.  Still can't understand what's wrong..

**Comment by domenukk** [comment, marked as answer]
It'll be very hard to tell what's going on. Maybe you can try to figure out what's going on using strace like so:
```sh
strace -tt -yy -y -f -e trace=openat,open,read,write,pipe,socket,dup2,clone,close -s 10000 -o /tmp/strace.log ./tool
```

**Comment by sk3pta** [comment]
I had a problem with Makefile xD Thank you!
